kwasm-lemmas.md

```diff
@@ -102,12 +112,12 @@ x = m * q + r, for a unique q and r s.t. 0 <= r < m
       [simplification]
 
     rule #wrap(N, (X <<Int M) +Int Y) => #wrap(N, Y)
-      requires 0 <=Int N
+      requires 0 <=Int M
        andBool N <=Int M
       [simplification]
 
     rule #wrap(N, Y +Int (X <<Int M)) => #wrap(N, Y)
-      requires 0 <=Int N
+      requires 0 <=Int M
        andBool N <=Int M
       [simplification]
```

Comment:
You might consider inspecting some of the arithmetic lemmas used in KEVM for handling the commutative cases: https://github.com/kframework/evm-semantics/blob/master/tests/specs/lemmas.k#L290

Reply:
Those are some nice features I wasn't aware of! That's great!

Is it alright if I make that a separate issue to deal with later? We have plenty of these duplicated lemmas so we should do one big overhaul.

Reply:
I'd prefer to land this (to land the do-balance spec) without introducing more changes here than necessary.